src/main/java/com/solvd/films/service/service/FilmServiceImpl.java

```diff
@@ -41,6 +41,15 @@ public Mono<Film> update(Long id, Integer amount) {
         }).flatMap(filmRepository::save);
     }
 
+    @Override
+    public Mono<Film> updateStatus(Long id) {
+        Mono<Film> filmFromDB = findById(id);
```

Comment:
Its better to chain the logic.

Reply:
Do you mean changing to update(Entity)? So first method with (Long id, Integer amount) is called when users buy tickets to decrease amount of remaining tickets. And the updateStatus method is the goal of message exchanging. Ticket Service sends the message, that all tickets are sold. Ad then in Film Service after getting this message status isSold becomes true with the help of this method.

Reply:
@vpetrovaa 
I mean:
```java
return
   findById(id)
   .map(filmUpdated -> {
            filmUpdated.setIsSold(true);
            return filmUpdated;
        }).flatMap(filmRepository::save);
```

while you doing this:
```java
Mono<Film> filmFromDB = findById(id);
return  filmFromDB.map(filmUpdated -> {
            filmUpdated.setIsSold(true);
            return filmUpdated;
        }).flatMap(filmRepository::save);
```